### PR TITLE
setup documentation generation

### PR DESCRIPTION
**Pull request by chrisantonellis**
* setup documentation generation
* setup hosted documentation site at https://wayfair-incubator.github.io/extra-model/site
* wrote documentation for how to generate + develop project documentation

### CODE REVIEW

File: docs/documentation.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Run the following command to render the documentation as `HTML` to the `/site` directory.
+
+```bash
+docker-compose run --rm mkdocs build
+```
+
+This directory is ignored in all branches except `gh-pages` branch. Use the following instructions to commit changes to documentation:
+
+1. Run `docker-compose run --rm mkdocs build` from the development branch.
+2. Run `git checkout gh-pages` to switch to the `gh-pages` branch and commit changes to the `/site` directory on this branch.
```
Comment (romatik) [inline docs/documentation.md:26]: does this mean that we need to have this branch setup?

Comment (chrisantonellis) [reply docs/documentation.md:26]: yes. i'll update the docs to mention pulling this branch down first 👍

File: docs/reference.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Reference
```
Comment (romatik) [inline docs/reference.md:1]: is there a way to include all functions here automatically? Or it's manual?

**Review by romatik — APPROVED** [review]

File: docs/reference.md (hunk 2 above)
Comment (chrisantonellis) [reply docs/reference.md:1]: its manual so we have the ability to exclude, annotate etc. as not all functions are intended to be public, not everything is required to be documented. its up to us!

**Pull request merged**